narwhals/expression.py

```diff
@@ -2375,13 +2375,58 @@ def sum(*columns: str) -> Expr:
     """
     Instantiate an expression representing the sum of one or more columns, similar to `polars.sum`.
     """
+
     return Expr(lambda plx: plx.sum(*columns))
 
 
 def mean(*columns: str) -> Expr:
     """
-    Instantiate an expression representing the mean of one or more columns, similar to `polars.mean`.
+    Get the mean value
+
+
+    Note:
+        Syntactic sugar for ``nw.col('name').mean()``
+
+
+    Parameters:
+        *names
+            Name(s) of the columns to use in the aggregation function
+
+
+    Examples:
+        >>> import pandas as pd
+        >>> import polars as pl
+        >>> import narwhals as nw
+        >>> df_pl = pl.DataFrame({"a": [1, 8, 3]})
+        >>> df_pd = pd.DataFrame({"a": [1, 8, 3]})
+
+
+    We define a dataframe agnostic function:
+
+
+        >>> def func(df_any):
+        ...     df = nw.from_native(df_any)
+        ...     df = df.select(nw.mean('a'))
+        ...     return nw.to_native(df)
+
+
+    We can then pass either pandas or Polars to `func`:
+
+
+        >>> func(df_pd)
```

Comment:
there's a lot of extra whitespace here - could we just have a single newline where necessary please?